### PR TITLE
Add platform profiling

### PR DESCRIPTION
**Pull request by jelovirt**
Add platform profiling controls, triggered by `multi-platform` class. This PR adopts changes from dita-ot/website#408.

Example output:

-   https://preview-pr-300--dita-ot.netlify.app/dev/topics/build-using-dita-command.html
-   https://preview-pr-300--dita-ot.netlify.app/dev/topics/determining-version-of-ditaot.html
-   https://preview-pr-300--dita-ot.netlify.app/dev/topics/using-docker-images.html

### CODE REVIEW

**Comment by jelovirt** [comment]
With https://preview-pr-300--dita-ot.netlify.app/dev/topics/determining-version-of-ditaot.html the issue is that the content of the choice table is inline content. In this case the output would look considerably better if the choice table content was block content, such as `<codeblock>`.

File: topics/using-docker-images.dita [hunk 1]
```diff
@@ -78,7 +78,7 @@
             <cmd>On the command line, run the <cmdname>docker</cmdname> command to log in with your GitHub
               credentials.</cmd>
             <stepxmp>
-              <codeblock><cmdname>docker</cmdname> login docker.pkg.github.com -u <varname>USERNAME</varname> -p <varname>PASSWORD/TOKEN</varname></codeblock>
+              <codeblock outputclass="language-bash"><cmdname>docker</cmdname> login docker.pkg.github.com -u <varname>USERNAME</varname> -p <varname>PASSWORD/TOKEN</varname></codeblock>
```
[the remark this answers is not in the document]
Comment (jelovirt) [reply topics/using-docker-images.dita:81]: Using `language-bash` is actually for the multi-platform code here. It allows us to rewrite `\` into `^` for multi-line commands.

[the remark this answers is not in the document]
Comment (jelovirt) [reply topics/using-docker-images.dita:81]: In order to keep the original styling, we could use some other class name to trigger to bash processing for our own code and not use something that triggers Prism.js.

File: topics/using-docker-images.dita [hunk 2]
```diff
@@ -107,13 +107,15 @@
               <li><option>-v</option> displays build progress messages with verbose logging</li>
             </ul>
           </p>
-          <p>On Windows, if your <filepath>Users</filepath> directory is on the <filepath>C:\</filepath> drive, use
+          <div platform="windows">
+            <p>On Windows, if your <filepath>Users</filepath> directory is on the <filepath>C:\</filepath> drive, use
               <filepath>/c/Users/…</filepath> to map the host directory:</p>
-          <codeblock>C:\Users\username> <cmdname>docker</cmdname> run -it \
+            <codeblock outputclass="language-batch">C:\Users\username> <cmdname>docker</cmdname> run -it \
```
[the remark this answers is not in the document]
Comment (jelovirt) [reply topics/using-docker-images.dita:113]: Here we have the same issue as with `language-bash`: Prism.js removes existing markup. So, maybe here too we should use some non-Prism.js class name instead.

File: topics/using-docker-images.dita [hunk 3]
```diff
@@ -107,13 +107,15 @@
...
   -v /c/Users/<varname>username</varname>/source:/src docker.pkg.github.com/dita-ot/dita-ot/dita-ot:<keyword keyref="maintenance-version"/> \
```
[the remark this answers is not in the document]
Comment (jelovirt) [reply topics/using-docker-images.dita:114]: When the source is intended to be multi-platform, IMO we should author in Bash and let the JS code rewrite it for multi-platform toggle. When the source is just Batch, it's better to actually write as Batch.

File: topics/increasing-the-jvm.dita [hunk 4]
```diff
@@ -45,7 +45,7 @@
             <choptionhd>Platform</choptionhd>
             <chdeschd>Command</chdeschd>
           </chhead>
-          <chrow platform="unix">
+          <chrow platform="linux mac" outputclass="multi-platform">
```
[the remark this answers is not in the document]
Comment (jelovirt) [reply topics/increasing-the-jvm.dita:48]: I will have to hard code the subject scheme for now, and then maybe convert the subject scheme into JSON so JS code could read it.

File: .github/workflows/preview.yml [hunk 5]
```diff
@@ -35,6 +35,7 @@ jobs:
         uses: actions/checkout@v2
         with:
           repository: dita-ot/website
+          ref: feature/platform-selector
```
Comment (jelovirt) [inline .github/workflows/preview.yml:38]: Do not merge this

**Comment by jelovirt** [comment]
> *    The backslash replacement doesn't seem to work consistently.
>     See Linux and macOS & Windows tabs in build-using-dita-command.html.

Fixed

>  *   The anchor IDs for Linux and macOS & Windows tabs seem to be auto-generated on each pageload with a v prefix & arbitrary numeric value. If the same page is loaded multiple times, the tab IDs are different each time. Any chance to use something more stable and semantic for these IDs (like #unix & #windows), so links to each tab would be persistent?

Fixed to code to not change the URL when platform is selected. It was a bug and the ID should not be in the URL.

> *    Also not sure how helpful it is to replace backslashes in codeblocks, but not in other filepaths, like the tip in https://preview-pr-300--dita-ot.netlify.app/dev/topics/determining-version-of-ditaot.html

This is a case where the `multi-platform` class is on a choicetable, but the note is out side the table. In order to change the filepath in the note, you'd have to put the `multi-platform` into e.g. the parent task `steps`.

> *   In that same file, the JavaScript changes are eating the space between the dita command and the double-hyphen option syntax, so dita --version appears as dita--version

Fixed.

**Comment by jelovirt** [comment]
It seems we don't actually use `language-bash` class name in our content, because it removes all existing markup.

What we could do, is disable `language-bash` with Prism.js and reuse the class name to add metadata that multi-platform code needs. Then, if we at some point find a code highlighter that retains nested markup, we can move to that.

**Comment by infotexture** [comment]
> What we could do, is disable `language-bash` with Prism.js and reuse the class name to add metadata that multi-platform code needs. Then, if we at some point find a code highlighter that retains nested markup, we can move to that.

Wouldn't it just be easier to use a different output class for our own purposes?
That would make it easier for authors to keep track of which class does what. 

Any `language-*` class is for syntax highlighting, etc. That and other permissible values are defined (and documented) in the subject scheme: https://github.com/dita-ot/docs/blob/develop/resources/subjectscheme-outputclass.ditamap

**Comment by jelovirt** [comment]
What would be good markup for defining element content to be either Shell (Bash) or Batch? Just go with `@outputclass` or  something like ``props="language(bash)"`. Is `lang-bash` too close? It's really the exactly the same information, you define the content language.

**Comment by infotexture** [comment]
> What would be good markup for defining element content to be either Shell (Bash) or Batch? Just go with `@outputclass` or something like ``props="language(bash)"`. Is `lang-bash` too close? It's really the exactly the same information, you define the content language.

I'd stick with output classes, and maybe use something explicit but related, like `syntax-bash` & `syntax-batch` to trigger our own JavaScript processing. 
 
> The `language-xxx` class name seems to be what HTML5 suggests using https://html.spec.whatwg.org/#the-code-element

Yes, that's why we (and Prism) chose it. That way if we switch highlighters at some point, there's a good chance we won't have to modify all the classes to something proprietary.

**Comment by jelovirt** [comment]
I'll add `syntax-bash` and `syntax-batch` support 👍

**Comment by jelovirt** [comment]
```

> 
> 
>     The first codeblock in https://preview-pr-300--dita-ot.netlify.app/dev/topics/build-using-dita-command.html doesn't seem to benefit from the substitution

There seems to be a bug in OT HTML5 code for steps processing. When a task has only a single step, it outputs the step without an ordered list wrapper. The bug is that `@outputclass` in `<steps>` is not output anywhere and thus we can't trigger the site JS processing. We can get this working when HTML5 plug-in has been fixed.

>     The final note in https://preview-pr-300--dita-ot.netlify.app/dev/topics/using-docker-images.html rewrites slashes on the Windows tab even where they should be preserved:

This is a case where `<filepath>` should be preserved. One option to fix this is to add something to disable separator rewrite

```xml
<filepath outputclass="preserve-separator">/opt/app/bin/</filepath>
```

We could also use the inverse, to add an `@outputclass` to enable automatic separator rewrite. Because retaining the separator is the less common case, I think we should adopt the former.

File: .github/workflows/preview.yml (hunk 5 above)
Comment (infotexture) [reply .github/workflows/preview.yml:38]: Shall we drop https://github.com/dita-ot/docs/pull/300/commits/667a82d29a234a331a3d3362d6346d3fc0db9855 with an interactive rebase before merging, or would you prefer to squash?

**Comment by jelovirt** [comment]
Rebased to remove the GitHub Actions configuration changes that used a feature branch of the website repo.

**Pull request merged**